hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientCacheHelper.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.cache.impl;
+
+import com.hazelcast.cache.impl.CacheProxyUtil;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.CacheCreateConfigCodec;
+import com.hazelcast.client.impl.protocol.codec.CacheGetConfigCodec;
+import com.hazelcast.client.impl.protocol.codec.CacheManagementConfigCodec;
+import com.hazelcast.client.spi.impl.ClientInvocation;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.core.Member;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.serialization.SerializationService;
+import com.hazelcast.util.ExceptionUtil;
+import com.hazelcast.util.FutureUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+final class ClientCacheHelper {
+
+    /**
+     * Thread-local cache config is used for passing created cache config indirectly
+     * from {@link com.hazelcast.client.cache.impl.HazelcastClientCacheManager}
+     * to {@link com.hazelcast.client.cache.impl.ClientCacheProxyFactory}
+     * to be used by {@link com.hazelcast.client.spi.ProxyManager} while creating client side cache proxy.
+     */
+    private static final ThreadLocal<CacheConfig> THREAD_LOCAL_CACHE_CONFIG = new ThreadLocal<CacheConfig>();
+
+    private ClientCacheHelper() {
+    }
+
+    static CacheConfig getThreadLocalCacheConfig() {
+        return THREAD_LOCAL_CACHE_CONFIG.get();
+    }
+
+    static void setThreadLocalCacheConfig(CacheConfig cacheConfig) {
+        THREAD_LOCAL_CACHE_CONFIG.set(cacheConfig);
+    }
+
+    static void removeThreadLocalCacheConfig() {
+        THREAD_LOCAL_CACHE_CONFIG.remove();
+    }
+
+    static <K, V>  CacheConfig<K, V> getCacheConfig(HazelcastClientInstanceImpl client,
+                                                    String cacheName, String simpleCacheName) {
+        ClientMessage request = CacheGetConfigCodec.encodeRequest(cacheName, simpleCacheName);
+        try {
+            int partitionId = client.getClientPartitionService().getPartitionId(cacheName);
```

Comment:
@bilalyasar `getNameWithPrefix()` of what? `cacheName` in here is the full cache name which contains also prefixes